frontend/src/pages/TripFormPage.tsx

```diff
@@ -1,15 +1,56 @@
 import "./TripFormPage.css";
+import LocationMap from "../components/LocationMap";
+import AddLocationForm from "../components/AddLocationForm";
+import Trip from "../types/Trip";
+import {useState} from "react";
+import {useNavigate} from "react-router-dom";
+import Api from "../api/Api";
+import Location from "../types/Location";
 
 export default function TripFormPage() {
-    // show input for title
-    // show map with markers for all added locations
-    // show AddLocationForm
-    // save button -> on click -> POST to api
-    // if post successful -> navigation to DashboardPage ?
+
+    const [trips, setTrips] = useState<Trip[]>([]);
+    const [locations, setLocations] = useState<Location[]>([]);
+    const [trip, setTrip] = useState<Trip>({title: "", tripTimestamp: "", locations: []});
+    const navigate =useNavigate();
+
+    const onSave = (() => {
+        (async () => {
+            const response:Trip = await Api.postTrip(trip);
+            setTrips([...trips,response]);
+            navigate("/");
+        })();
+    });
+
+    const onAdd = (location : Location)=> {
+        setLocations([...locations,location]);
+        setTrip({...trip, locations: locations})
```

Comment:
Zum Verständnis: Was ist denn der Vorteil, wenn man Locations und Trip hier aufsplittet? 
Ist das dann einfacher zu handhaben, als wenn man nur einen trip State hat und keinen separaten state für die locations?

Reply:
ich dachte es wäre einfacher zu verstehen. aber geht auch mit nur einen trip State. soll ich den ändern?

Reply:
Hab da keine Präferenz - würde ich dir überlassen :)

Reply:
Ich glaube, wir sollten das doch umbauen, dass locations kein separater State ist. 

Hab lokal mal kurz nen log eingebaut: 
```
const onAdd = (location : Location)=> {
        setLocations([...locations,location]);
        console.log("added location:", location, "locations are now:", locations);
        setTrip({...trip, locations: locations})
    }
```

Ausgabe in der Console ist dann beim hinzufügen einer location:
```
added location: {name: 'Kölner Dom', latitude: 50.941386546092225, longitude: 6.958270670147375} locations are now: []
```

sieht für mich so aus, als würde hier tatsächlich genau das Problem sein, dass die locations noch nicht geupdated sind über `setLocations`, wenn sie im trip benutzt werden. 

Wenn man den separaten locations state rausnimmt und 
```
const onAdd = (location : Location)=> {
        setTrip({...trip, locations: [...locations, location]})
    }
```
macht, funktioniert auch das speichern der location. 

Hier auch nochmal der Artikel, in dem ich davon gelesen hab, dass useState asynchronous is: 
https://dev.to/shareef/react-usestate-hook-is-asynchronous-1hia

Reply:
ok dann haben wir das Problem gefunden. Danke!

Reply:
gefixt